### PR TITLE
Handler(Enhancer)Definitions configuration.

### LINKED ISSUE #52: Update HandlerEnhancerDefinition and HandlerDefinition documentation
Update `HandlerEnhancerDefinition` and `HandlerDefinition` documentation in the 'Advanced customization' section.
See [#556](https://github.com/AxonFramework/AxonFramework/issues/556) for the implementation.

### PR DESCRIPTION
**Pull request by m1l4n54v1c**
Resolves #52

### CODE REVIEW

File: part-iv-advanced-tuning/advanced-customizations.md [hunk 1]
```diff
@@ -105,6 +105,8 @@ You can then sort these handlers based on their annotations by using the `annota
 For your HandlerEnhancer to run you'll need to create a `META-INF/services/org.axonframework.messaging.annotation.HandlerEnhancerDefinition` file containing
 the fully qualified class name of the handler enhancer you have created.
 
+> **Note** Since 3.3 Axon version it is possible to configure `HandlerDefinition` with Axon `Configuration`. If you are using Spring Boot defining `HandlerDefintion`s and `HandlerEnhancerDefinition`s as beans is sufficient (Axon autoconfiguration will pick them up and configure within Axon `Configuration`).
+
```
Comment (smcvb) [inline part-iv-advanced-tuning/advanced-customizations.md]: A code sample to configure, albeit minimal, I think is a good addition.

File: part-iv-advanced-tuning/advanced-customizations.md [hunk 2]
```diff
@@ -105,6 +105,8 @@ You can then sort these handlers based on their annotations by using the `annota
 For your HandlerEnhancer to run you'll need to create a `META-INF/services/org.axonframework.messaging.annotation.HandlerEnhancerDefinition` file containing
 the fully qualified class name of the handler enhancer you have created.
 
+> **Note** Since 3.3 Axon version it is possible to configure `HandlerDefinition` with Axon `Configuration`. If you are using Spring Boot defining `HandlerDefintion`s and `HandlerEnhancerDefinition`s as beans is sufficient (Axon autoconfiguration will pick them up and configure within Axon `Configuration`).
```
Comment (smcvb) [inline part-iv-advanced-tuning/advanced-customizations.md]: I don't think we have to note that this is possible as off framework version 3.3, as users will only see this if they have RefGuide version 3.3 (or higher) opened up. It should go without saying.

Comment (m1l4n54v1c) [reply part-iv-advanced-tuning/advanced-customizations.md]: In future versions (3.4, 3.5, ...) you'd have to go through the past releases in order to find in which one this was added. We can remove it if you think it's not a valuable info.

Comment (smcvb) [reply part-iv-advanced-tuning/advanced-customizations.md]: I think this reasoning would require us to retroactively add the version number to every feature which is added to the framework.

And why would you need to check in future versions when something was added?
The javadoc of a given class should provide the version number already, which I personally think is sufficient.

File: part-iv-advanced-tuning/advanced-customizations.md [hunk 3]
```diff
@@ -154,6 +153,12 @@ public class MethodCommandHandlerDefinition implements HandlerEnhancerDefinition
 
 To skip all handling of the handler then just throw an exception.
 
+Since 3.3 Axon version it is possible to configure `HandlerDefinition` with Axon `Configuration`. If you are using Spring Boot defining `HandlerDefintion`s and `HandlerEnhancerDefinition`s as beans is sufficient (Axon autoconfiguration will pick them up and configure within Axon `Configuration`).
+
+```java
+configurer.registerHandlerDefinition(c -> new MethodCommandHandlerDefinition());
```
Comment (smcvb) [inline part-iv-advanced-tuning/advanced-customizations.md]: Might be nice to state `Configurer configurer = ...` or provide it as a parameter to an example configure method, as I think it is beneficial to be a bit more specific.

File: part-iv-advanced-tuning/advanced-customizations.md [hunk 4]
```diff
@@ -154,6 +153,12 @@ public class MethodCommandHandlerDefinition implements HandlerEnhancerDefinition
 
 To skip all handling of the handler then just throw an exception.
 
+Since 3.3 Axon version it is possible to configure `HandlerDefinition` with Axon `Configuration`. If you are using Spring Boot defining `HandlerDefintion`s and `HandlerEnhancerDefinition`s as beans is sufficient (Axon autoconfiguration will pick them up and configure within Axon `Configuration`).
+
+```java
```
Comment (smcvb) [inline part-iv-advanced-tuning/advanced-customizations.md]: For all the other configuration code samples, we also provide the `Spring Boot Auto Configure` situation.
I know that's just showing a bean creation method, but nonetheless I think it is fair to be consistent on this.

**Pull request merged**